### ISSUE TITLE
Add ElasticArray{T,N,U} convert and constructor

### ISSUE THREAD

**Issue by platawiec** [JuliaArrays/ElasticArrays.jl#12]
Currently the constructors and converters don't support `ElasticArray{T, N, U}`. This is a problem when, for instance, using the type from a parametric struct. Consider:

```
using ElasticArrays

struct A{T}
    a::T
end
function f(a::A{T}, x) where T return convert(T, x) end

a = A([1])

f(a, [2])
ERROR: MethodError: Cannot `convert` an object of type Array{Int64,1} to an object of type ElasticArray{Int64,1,0}
```

**Comment by oschulz** [comment]
@platawiec , your example doesn't seem to actually use `ElasticArray`s. But I guess you mean something like this: #13

**Comment by oschulz** [comment]
@platawiec, if #13 works for you, it would be time for a new release of ElasticArrays anyhow (to add a "Project.toml" and remove "REQUIRE").

**Comment by platawiec** [comment]
Yes, of course, should have written `a = A(ElasticArray([1]))` above. #13 is exactly what I had in mind.

**Comment by oschulz** [comment]
Ok, then. :-)

Closed by #13.
